apps/passport-server/src/services/devconnectPretixSyncService.ts

```diff
@@ -153,22 +161,193 @@ export class DevconnectPretixSyncService {
     });
   }
 
+  /**
+   * Validate that an event's settings match our expectations.
+   * These settings correspond to the "Ask for email addresses per ticket"
+   * setting in the Pretix UI being set to "Ask and require input", which
+   * is mandatory for us.
+   */
+  private validateEventSettings(
+    settings: DevconnectPretixEventSettings
+  ): boolean {
+    if (
+      settings.attendee_emails_asked === true &&
+      settings.attendee_emails_required === true
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Validate that an item/products settings match our expectations.
+   * These settings correspond to the product being of type "Admission",
+   * "Personalization" being set to "Personalized ticket", and
+   * "Generate tickets" in the "Tickets & Badges" section being set to
+   * "Choose automatically depending on event settings" in the Pretix UI.
+   */
+  private validateEventItem(item: DevconnectPretixItem): boolean {
+    if (
+      item.admission === true &&
+      item.personalized === true &&
+      (item.generate_tickets === null || item.generate_tickets === undefined)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Check all of the API responses for an event before syncing them to the
+   * DB.
+   */
+  private checkEventData(
+    eventData: EventData,
+    eventConfig: DevconnectPretixEventConfig
+  ): void {
+    const { settings, items } = eventData;
+    const activeItemIdSet = new Set(eventConfig.activeItemIDs);
+
+    // We want to make sure that we log all errors, so we collect everything
+    // and only throw an exception once we have found all of them.
+    const errors = [];
+
+    if (!this.validateEventSettings(settings)) {
+      errors.push(
+        `Event settings for "${eventData.eventInfo.name}" (${eventData.eventInfo.slug}) are invalid`
+      );
+    }
```

Comment:
I think it would be good to report what in particular is invalid about these - perhaps `validateEventSettings` and `validateEventItem` could return an array of error strings that specifically refer to which setting is invalid. I'm just thinking that if I encounter this error in some sort of alert, I would prefer for it to be actionable without further debugging.